src/tools/PrizeTierController/SavePrizeTiersModal/DrawIdForm.tsx

```diff
@@ -0,0 +1,74 @@
+import { ErrorMessage } from '@components/ErrorMessage'
+import { StyledInput } from '@components/Input'
+import { Label } from '@components/Label'
+import { useEffect } from 'react'
+import { FieldErrorsImpl, useForm, UseFormRegister } from 'react-hook-form'
+
+export const DrawIdForm = (props: {
+  onChange: (drawId: number) => void
+  defaultValues: { drawId: string }
+  minDrawId: number
+}) => {
+  const { onChange, defaultValues, minDrawId } = props
+  const {
+    register,
+    watch,
+    formState: { errors, isValid }
+  } = useForm<{ drawId: string }>({
+    mode: 'onChange',
+    defaultValues,
+    shouldUnregister: true
+  })
+
+  const drawId = watch('drawId')
+
+  useEffect(() => {
+    const numDrawId = Number(drawId)
+    if (Number.isInteger(numDrawId) && numDrawId > minDrawId) {
+      onChange(numDrawId)
+    }
+  }, [drawId])
+
+  return (
+    <form className='flex flex-col' autoComplete='off'>
+      <DrawIdInput minDrawId={minDrawId} errors={errors} register={register} />
+    </form>
+  )
+}
+
+const DrawIdInput = (props: {
+  minDrawId: number
+  errors: FieldErrorsImpl<{ drawId: string }>
+  register: UseFormRegister<{ drawId: string }>
+}) => {
+  const { minDrawId, errors, register } = props
+
+  return (
+    <div>
+      <Label className='uppercase' htmlFor='drawId'>
+        Draw ID
+      </Label>
+      <StyledInput
+        id='drawId'
+        invalid={!!errors.drawId}
+        className='w-full'
+        {...register('drawId', {
+          required: {
+            value: true,
+            message: 'Draw ID is required'
+          },
+          validate: {
+            isValidInteger: (v) => Number.isInteger(Number(v)) || 'Invalid Draw ID',
```

Comment:
I believe there's a prop for `react-hook-form` so it knows this value is an integer. You should get these validations for free. Check out their docs!

Reply:
They do have a `valueAsNumber`, but it pretty much just casts it to a `Number` or returns `NaN`. Almost all the inputs here are integers only and would require further validation anyways. They don't have any integer-specific validation as far as I could tell.